### ISSUE TITLE
Kubernetes 1.7 is there and our Vagrant kubeadm setup does not work anymore

### ISSUE THREAD

**Issue by rmohr** [kubevirt/kubevirt#286]
Yes it is that time again :)

Kubeadm tries to deploy Kubernets 1.7 after a fresh `vagrant up`. The master comes up properly, but the node can't register with the provided token.

The error message is:

```
==> node0: [discovery] Trying to connect to API Server "192.168.200.2:6443"
==> node0: [discovery] Created cluster-info discovery client, requesting info from "https://192.168.200.2:6443"
==> node0: [discovery] Failed to connect to API Server "192.168.200.2:6443": there is no JWS signed token in the cluster-info ConfigMap. This token id "abcdef" is invalid for this cluster, can't connect
```
There were some changes regarding to the token and bootstrapping.

**Comment by rmohr** [comment]
I can start working on this shortly, but if someone has time right now, feel free to pick it.

**Comment by rmohr** [comment]
Got it: https://github.com/kubernetes/kubeadm/issues/335

**Comment by rmohr** [comment]
Provided a temporary fix in #288

**Comment by stu-gott** [comment]
As mentioned in https://github.com/kubernetes/kubernetes/pull/48480 the root cause of this appears to be a race condition (already merged into upstream git). In order to get Kubernetes 1.7 working before that patch hits, I simply recreated my vagrant nodes until everything came up properly. It took two re-creates for me. YMMV
